scripts/update_package_versions.py

```diff
@@ -260,66 +289,79 @@ def get_public_ids_to_update() -> Set[PackageId]:
             deployed_public_id = public_id_in_registry(type_, name)
             difference = minor_version_difference(current_public_id, deployed_public_id)
             # check if the public ids of the local package and the package in the registry are already the same.
+            package_info = f"Package `{name}` of type `{type_}`"
+            public_id_info = f"current id `{current_public_id}` and deployed id `{deployed_public_id}`"
             if difference == 0:
-                print(
-                    "Package `{}` of type `{}` needs to be bumped!".format(name, type_)
-                )
+                print(f"{package_info} needs to be bumped!")
                 result.add(PackageId(type_[:-1], current_public_id))
             elif difference == 1:
-                print(
-                    "Package `{}` of type `{}` already at correct version!".format(
-                        name, type_
-                    )
-                )
+                print(f"{package_info} already at correct version!")
                 continue
             else:
-                print(
-                    "Package `{}` of type `{}` has current id `{}` and deployed id `{}`. Error!".format(
-                        name, type_, current_public_id, deployed_public_id
-                    )
-                )
+                print(f"{package_info} has {public_id_info}. Error!")
                 sys.exit(1)
     return result
 
 
-def _extract_prefix(public_id: PublicId) -> Tuple[str, str]:
-    """Extract (author, package_name) from public id."""
-    return public_id.author, public_id.name
-
-
-def _get_ambiguous_public_ids(
-    all_package_ids_to_update: Set[PackageId],
-) -> Set[PublicId]:
+def _get_ambiguous_public_ids() -> Set[PublicId]:
     """Get the public ids that are the public ids of more than one package id."""
+    all_package_ids = get_all_package_ids()
```

Comment:
this is important... ambiguous public ids must be computed from all the packages in the repo, not just the one being updated.